benchopt/runner.py

```diff
@@ -205,6 +205,77 @@ def get_next(x): return max(x / rho, MIN_TOL)
     return curve, status
 
 
+class Callback:
+
+    def __init__(self, objective, max_iter, deadline, meta):
+        """Get callback function and store informations for the curve.
+
+        Parameters
+        ----------
+        objective : instance of BaseObjective
+            The objective to minimize.
+        max_iter : int
+            Maximum number of iterations to run for.
+        deadline : float
+            Deadline after which to stop the computations. This will be
+            used to respect the timeout for each solver.
+        meta : dict
+            Metadata passed to store in Cost results.
+            Contains objective and data names, problem dimension, etc.
+        """
+
+        self.objective = objective
+        self.max_iter = max_iter
+        self.deadline = deadline
+        self.meta = meta
+        self.info = get_sys_info()
+        self.curve = []
+        self.cb_status = {
+                'status': 'unfinished',
+                "it": 0,
+                'next_stopval': 0,
+                'delta_t': 0,
+                'time': time.perf_counter(),
+        }
+
+    def __call__(self, beta_hat_i):
+        t0 = time.perf_counter()
+        self.cb_status["delta_t"] += t0 - self.cb_status["time"]
+        if self.cb_status["it"] == self.cb_status["next_stopval"]:
+            objective_dict = self.objective(beta_hat_i)
+            self.curve.append(dict(
+                    **self.meta, stop_val=self.cb_status["it"],
+                    time=self.cb_status["delta_t"],
+                    **objective_dict, **self.info
+            ))
+
+            if self.deadline is not None and time.time() > self.deadline:
+                self.cb_status["status"] = 'timeout'
+                return False
+            if self.cb_status["it"] == self.max_iter:
+                return False
+
+            self.cb_status["next_stopval"] = min(
+                get_next(self.cb_status["next_stopval"]), self.max_iter
```

Comment:
```suggestion
                get_next(self.cb_status["next_stopval"], strategy="iteration"), self.max_iter
```
to be explicit that Callback only works with iterations